packages/flutter/lib/src/foundation/annotations.dart

```diff
@@ -0,0 +1,107 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+/// A category with which to annotate a class, for documentation
+/// purposes.
+///
+/// A category is usually represented as a section and a subsection, each
+/// of which is a string. The project to which the class belongs defines the
+/// categories used for classes in that project. For example, Flutter defines
+/// categories like "Basic/Buttons" and "Material Design/Buttons".
+///
+/// A class can have multiple categories.
+///
+/// ## Sample code
+///
+/// ```dart
+/// /// A copper coffee pot, as desired by Ben Turpin.
+/// /// ...documentation...
+/// @Category(const <String>['Pots', 'Coffee'])
+/// @Category(const <String>['Copper', 'Cookware'])
+/// @DocumentationIcon('https://example.com/images/coffee.png')
+/// @Summary('A proper cup of coffee is made in a proper copper coffee pot.')
+/// class CopperCoffeePot {
+///   // ...code...
+/// }
+/// ```
+///
+/// See also:
+///
+///  * [DocumentationIcon], which is used to give the URL to an image that
+///    represents the class.
+///  * [Summary], which is used to provide a one-line description of a
```

Comment:
maybe add "optional", because many API docs will have a sufficient one-line doc in their doc comments. I assume @Summary is only required when the existing one-line doc isn't sufficient for generated catalogs, etc.